src/model/openthaigpt_pretraining_model/pl_trainer/optimizer/scheduler.py

```diff
@@ -0,0 +1,138 @@
+# Implement from https://github.com/mlfoundations/open_clip/blob/main/src/training/scheduler.py
+import numpy as np
+import torch
+import math
+
+INITIAL_LR = "initial_lr"
+
+
+def _warmup_lr(base_lr, warmup_length, step):
```

Comment:
Can we have type hint for this function?

Reply:
Done